docs/design/001-cli-config-structure.md

```diff
@@ -0,0 +1,115 @@
+
+The goal of the second section of this document is to establish a clear format, expectation and language around the structure of the config file and its contents along with how that maps to plugins and features of those plugins.
+
+## Proposal
+
+First lets create some common definitions to remove ambiguity surrounding language with regards to the aspect cli.
+
+## Definitions
+
+### Plugin
+
+For the aspect cli a plugin refers to something that exposes a grpc server this will typically be a binary.
+The definition of a plugin is not associated with any functionality inside that binary or anything else but simply that it exposes a grpc server and interacts though that server.
+
+Plugins can be turned on and off independently unless they are explicitly specified as a dependency of another plugin. This will be discussed further later on.
+
+### Feature
+
+A feature refers to some functionality that is exposed through a plugin. There can be multiple features in a single plugin.
+
+Features should be exposed independently inside of a plugin so that they can be turned on and off independently.
+
+## Config File Content Structure
+
+The aspect cli config should be broken down into different namespaces that map to each plugin. This structure can be nested so that features of plugins can also use their own namespace structure to control their own features and settings. The core aspect cli will only deal with the top level namespaces and turing those plugins on appropriately. Dealing with the rest is up to the creator of the plugin. It is recommended that the same namespace structure is used to control features of a plugin as well so that it is easy and consistent for users to understand a config file with multiple plugins and update them as needed.
+
+This should include a relevant namespace nested example with multiple plugins and features:
+
+It is also relevant to know that all features available through the config file should also be flags that can be passed to the aspect cli. One should also consider that features and settings in the config file should be modifiable through a command in the aspect cli that writes to the config file. This is just to say take care in naming and structuring your namespace so that commands are readable on their own.
+
+How to deal with plugin ordering and dependencies?
+This is the lest discussed section. Open for comment. Should we just add a common top level way for aspect cli core to understand which plugins are required for others to run? How does this work with a plugin that has multiple features? Can the feature say it needs the plugin or does the top level have to say it requires that plugin? What does the error message and error handling look like here? How do we make sure there is an obvious and actionable error message for a user?
```

Comment:
Are you looking for help with this part of the design then? Please reach out and be noisy and don't wait for someone to come ask questions in the doc.